File: app/controllers/Application.java
package controllers;

import actors.*;
import akka.actor.*;
import akka.actor.ActorRef;
import com.fasterxml.jackson.databind.JsonNode;
import play.libs.Akka;
import play.libs.F;
import play.mvc.Controller;
import play.mvc.Result;
import play.mvc.WebSocket;
import scala.Option;


/**
 * The main web controller that handles returning the index page, setting up a WebSocket, and watching a stock.
 */
public class Application extends Controller {

    public Result index() {
        return ok(views.html.index.render());
    }

    public WebSocket<JsonNode> ws() {
        return new WebSocket<JsonNode>() {
            public void onReady(final WebSocket.In<JsonNode> in, final WebSocket.Out<JsonNode> out) {
                // create a new UserActor and give it the default stocks to watch
                final ActorRef userActor = Akka.system().actorOf(Props.create(UserActor.class, out));
                
                // send all WebSocket message to the UserActor
                in.onMessage(new F.Callback<JsonNode>() {
                    @Override
                    public void invoke(JsonNode jsonNode) throws Throwable {
                        // parse the JSON into WatchStock
                        WatchStock watchStock = new WatchStock(jsonNode.get("symbol").textValue());
                        // send the watchStock message to the StocksActor
                        StocksActor.stocksActor().tell(watchStock, userActor);
                    }
                });

                // on close, tell the userActor to shutdown
                in.onClose(new F.Callback0() {
                    @Override
                    public void invoke() throws Throwable {
                        final Option<String> none = Option.empty();
                        StocksActor.stocksActor().tell(new UnwatchStock(none), userActor);
                        Akka.system().stop(userActor);
                    }
                });
            }
        };
    }

}

File: app/controllers/Application.scala
package controllers

import java.util.UUID
import javax.inject.Inject

import akka.actor.ActorSystem
import com.example.user.{User, UserDAO}
import play.api._
import play.api.mvc._

class Application @Inject() (userDAO: UserDAO, actorSystem:ActorSystem) extends Controller {

  private val ec = actorSystem.dispatchers.lookup("myapp.database-dispatcher")

  def index = Action.async {
    // Set up an execution context from the akka dispatchers library...

    val id = UUID.randomUUID.toString
    userDAO.create(User(id, "some@example.com")).map { rows =>
      Ok(views.html.index(s"Connected to database and created user $id"))
    }(ec)
  }

}
